File: backend/src/modules/exercicio/service/ExercicioService.js
const repository = require('../repository/ExercicioRepository');
class ExercicioService {
  async getAll() { return repository.findAll(); }
  async getById(id) { const e = await repository.findById(id); if (!e) { const err = new Error('Exercício não encontrado'); err.status = 404; throw err; } return e; }
  async create(data) { const Exercicio = require('../entity/Exercicio'); const e = new Exercicio(data); const errors = e.validate(); if (errors.length > 0) { const err = new Error(errors.join(', ')); err.status = 400; throw err; } return repository.create(e); }
  async delete(id) { return repository.remove(id); }
}
module.exports = new ExercicioService();

File: backend-spring/src/main/java/com/vitalativa/service/ExercicioService.java
package com.vitalativa.service;

import com.vitalativa.entity.Exercicio;
import com.vitalativa.repository.ExercicioRepository;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class ExercicioService {
    
    private final ExercicioRepository repository;
    
    public ExercicioService(ExercicioRepository repository) {
        this.repository = repository;
    }
    
    public List<Exercicio> findAll() {
        return repository.findAll();
    }
    
    public Exercicio findById(Integer id) {
        return repository.findById(id).orElse(null);
    }
    
    public Exercicio save(Exercicio exercicio) {
        return repository.save(exercicio);
    }
    
    public void delete(Integer id) {
        repository.deleteById(id);
    }
}